jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -196,12 +196,13 @@ object CodeWhispererUtil {
     //   for example, when user performs security scan or fetch code completion for the first time
     // Return true if need to re-auth, false otherwise
     fun promptReAuth(project: Project, isPluginStarting: Boolean = false): Boolean {
-        if (CodeWhispererService.hasReAuthPromptBeenShown()) return false
         if (!isCodeWhispererExpired(project)) return false
         val tokenProvider = tokenProvider(project) ?: return false
         return maybeReauthProviderIfNeeded(project, tokenProvider) {
             runInEdt {
-                notifyConnectionExpiredRequestReauth(project)
+                if (!CodeWhispererService.hasReAuthPromptBeenShown()) {
+                    notifyConnectionExpiredRequestReauth(project)
+                }
```

Comment:
I prefer adding `if (isCodeWhispererExpired(project)) return` directly at `CodeWhispererService.kt` right after where we invoke `promptReauth` 

The reasoning is that our goal here is to handle cases when
1. users have seen "reconnect" UI and didn't not take action to the expired connection
2. in such schenario, we want to block upcoming triggers from invoking the real service

I think the `promptReauth` the function name itself suggest its functionality is to determine if we need to prompt the UI but not to terminate the logic from proceeding.

Reply:
Ditto. I also feel like adding a seperate check out side of promptReAuth suit the common sense to me. @andrewyuq What's you thought on this issue?

Reply:
yes that works too, there are only 2 places calling this function, not a blocker.

Reply:
you can also leave it as is, I am fine with that. Either way, we need to test it. Try edit the access token file in the sso/cache, try 
1. edit access token expiration date
2. edit access token expiration date + refresh token 

that's it then we are good to go

Reply:
Add project.refreshDevToolTree() and did functional test:
Connection expired prompt show up once at restart IDE, also show up at the first time trigger recommendation.
Later when trigger recommendation at expired state, abort the request and show no message to user.

Reply:
> I prefer adding if (isCodeWhispererExpired(project)) return directly at CodeWhispererService.kt right after 
> where we invoke promptReauth

this works for me.

Reply:
`project.refreshDevToolTree()` inside `maybeReauthProviderIfNeeded` makes sense to me. Since in all "fail to refresh" cases, we are calling the callback inside `maybeReauthProviderIfNeeded`.

Reply:
status bar show expired already before adding refreshDevToolTree

Reply:
I think this is good to go.